sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/Manager.java

```diff
@@ -56,6 +58,11 @@ protected final void withResourceManager(ResourceManager resourceManager) {
         this.resourceManager = resourceManager;
     }
 
+    protected <T extends AzureConfigurable<?>> T withHttpPipeline(HttpPipeline httpPipeline, T azureConfigurable) {
```

Comment:
This will be part of the public API. Do we need this method to hide the authenticate method that uses the pipeline?

Reply:
As discussed, move this method into `AzureConfigurableImpl` and make it public static.